### ISSUE TITLE
Weird crash every morning

### ISSUE THREAD

**Issue by Christoph-AK** [poem-web/poem#174]
Hey there,
I am hosting a server running poem-openapi on AWS EC2. 

It frequently crashes early in the morning with following backtrace. Can someone help me make sense of this? 
As far as i can see the only line containing useful information is 9, but I don't understand what goes wrong.

```
[2022-01-20 05:17:26.796666] ERROR [log_panics] /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/log-panics-2.0.0/src/lib.rs:67: thread 'tokio-runtime-worker' panicked at 'called `Option::unwrap()` on a `None` value': /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/poem-1.2.36/src/route/router.rs:275
   0: log_panics::init::{{closure}}
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/log-panics-2.0.0/src/lib.rs:52:25
   1: std::panicking::rust_panic_with_hook
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/panicking.rs:610:17
   2: std::panicking::begin_panic_handler::{{closure}}
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/panicking.rs:500:13
   3: std::sys_common::backtrace::__rust_end_short_backtrace
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/sys_common/backtrace.rs:139:18
   4: rust_begin_unwind
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/panicking.rs:498:5
   5: core::panicking::panic_fmt
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/panicking.rs:107:14
   6: core::panicking::panic
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/panicking.rs:48:5
   7: core::option::Option<T>::unwrap
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/option.rs:746:21
      <poem::route::router::Route::internal_nest::Nest<E> as poem::endpoint::endpoint::Endpoint>::call::{{closure}}
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/poem-1.2.36/src/route/router.rs:275:26
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/future/mod.rs:80:19
   8: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/future/future.rs:119:9
      <alloc::boxed::Box<T> as poem::endpoint::endpoint::Endpoint>::call::{{closure}}
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/poem-1.2.36/src/endpoint/endpoint.rs:188:9
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/future/mod.rs:80:19
   9: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/future/future.rs:119:9
      <poem::route::router::Route as poem::endpoint::endpoint::Endpoint>::call::{{closure}}
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/poem-1.2.36/src/route/router.rs:329:17
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/future/mod.rs:80:19
  10: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/future/future.rs:119:9
      <poem::middleware::cors::CorsEndpoint<E> as poem::endpoint::endpoint::Endpoint>::call::{{closure}}
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/poem-1.2.36/src/middleware/cors.rs:337:24
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/future/mod.rs:80:19
  11: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/future/future.rs:119:9
      <poem::endpoint::map_to_response::MapToResponse<E> as poem::endpoint::endpoint::Endpoint>::call::{{closure}}
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/poem-1.2.36/src/endpoint/map_to_response.rs:21:9
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/future/mod.rs:80:19
  12: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/future/future.rs:119:9
      <alloc::sync::Arc<T> as poem::endpoint::endpoint::Endpoint>::call::{{closure}}
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/poem-1.2.36/src/endpoint/endpoint.rs:197:9
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/future/mod.rs:80:19
  13: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/future/future.rs:119:9
      poem::endpoint::endpoint::Endpoint::get_response::{{closure}}
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/poem-1.2.36/src/endpoint/endpoint.rs:43:9
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/future/mod.rs:80:19
  14: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/future/future.rs:119:9
      poem::server::serve_connection::{{closure}}::{{closure}}::{{closure}}
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/poem-1.2.36/src/server.rs:184:21
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/future/mod.rs:80:19
  15: <hyper::proto::h1::dispatch::Server<S,hyper::body::body::Body> as hyper::proto::h1::dispatch::Dispatch>::poll_msg
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.14.16/src/proto/h1/dispatch.rs:491:35
      hyper::proto::h1::dispatch::Dispatcher<D,Bs,I,T>::poll_write
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.14.16/src/proto/h1/dispatch.rs:297:43
      hyper::proto::h1::dispatch::Dispatcher<D,Bs,I,T>::poll_loop
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.14.16/src/proto/h1/dispatch.rs:161:21
      hyper::proto::h1::dispatch::Dispatcher<D,Bs,I,T>::poll_inner
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.14.16/src/proto/h1/dispatch.rs:137:16
      hyper::proto::h1::dispatch::Dispatcher<D,Bs,I,T>::poll_catch
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.14.16/src/proto/h1/dispatch.rs:120:28
  16: <hyper::proto::h1::dispatch::Dispatcher<D,Bs,I,T> as core::future::future::Future>::poll
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.14.16/src/proto/h1/dispatch.rs:424:9
      <hyper::server::conn::ProtoServer<T,B,S,E> as core::future::future::Future>::poll
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.14.16/src/server/conn.rs:1133:47
      <hyper::server::conn::upgrades::UpgradeableConnection<I,S,E> as core::future::future::Future>::poll
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/hyper-0.14.16/src/server/conn.rs:1337:30
  17: poem::server::serve_connection::{{closure}}
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/poem-1.2.36/src/server.rs:195:13
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/future/mod.rs:80:19
  18: poem::server::Server<L,A>::run_with_graceful_shutdown::{{closure}}::{{closure}}
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/poem-1.2.36/src/server.rs:146:33
      <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/future/mod.rs:80:19
  19: tokio::runtime::task::core::CoreStage<T>::poll::{{closure}}
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/core.rs:161:17
      tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/loom/std/unsafe_cell.rs:14:9
      tokio::runtime::task::core::CoreStage<T>::poll
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/core.rs:151:13
      tokio::runtime::task::harness::poll_future::{{closure}}
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/harness.rs:461:19
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/panic/unwind_safe.rs:271:9
      std::panicking::try::do_call
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/panicking.rs:406:40
      std::panicking::try
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/panicking.rs:370:19
      std::panic::catch_unwind
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/panic.rs:133:14
      tokio::runtime::task::harness::poll_future
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/harness.rs:449:18
      tokio::runtime::task::harness::Harness<T,S>::poll_inner
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/harness.rs:98:27
      tokio::runtime::task::harness::Harness<T,S>::poll
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/harness.rs:53:15
  20: tokio::runtime::task::raw::RawTask::poll
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/raw.rs:70:18
      tokio::runtime::task::LocalNotified<S>::run
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/mod.rs:343:9
      tokio::runtime::thread_pool::worker::Context::run_task::{{closure}}
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/thread_pool/worker.rs:420:13
      tokio::coop::with_budget::{{closure}}
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/coop.rs:102:9
      std::thread::local::LocalKey<T>::try_with
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/thread/local.rs:399:16
      std::thread::local::LocalKey<T>::with
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/thread/local.rs:375:9
  21: tokio::coop::with_budget
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/coop.rs:95:5
      tokio::coop::budget
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/coop.rs:72:5
      tokio::runtime::thread_pool::worker::Context::run_task
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/thread_pool/worker.rs:419:9
  22: tokio::runtime::thread_pool::worker::Context::run
  23: tokio::runtime::thread_pool::worker::run::{{closure}}
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/thread_pool/worker.rs:371:17
      tokio::macros::scoped_tls::ScopedKey<T>::set
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/macros/scoped_tls.rs:61:9
  24: tokio::runtime::thread_pool::worker::run
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/thread_pool/worker.rs:368:5
  25: tokio::runtime::thread_pool::worker::Launch::launch::{{closure}}
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/thread_pool/worker.rs:347:45
      <tokio::runtime::blocking::task::BlockingTask<T> as core::future::future::Future>::poll
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/blocking/task.rs:42:21
      tokio::runtime::task::core::CoreStage<T>::poll::{{closure}}
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/core.rs:161:17
      tokio::loom::std::unsafe_cell::UnsafeCell<T>::with_mut
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/loom/std/unsafe_cell.rs:14:9
  26: tokio::runtime::task::core::CoreStage<T>::poll
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/core.rs:151:13
  27: tokio::runtime::task::harness::poll_future::{{closure}}
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/harness.rs:461:19
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/panic/unwind_safe.rs:271:9
      std::panicking::try::do_call
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/panicking.rs:406:40
      std::panicking::try
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/panicking.rs:370:19
      std::panic::catch_unwind
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/panic.rs:133:14
      tokio::runtime::task::harness::poll_future
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/harness.rs:449:18
      tokio::runtime::task::harness::Harness<T,S>::poll_inner
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/harness.rs:98:27
      tokio::runtime::task::harness::Harness<T,S>::poll
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/harness.rs:53:15
  28: tokio::runtime::task::raw::RawTask::poll
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/raw.rs:70:18
      tokio::runtime::task::UnownedTask<S>::run
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/task/mod.rs:379:9
      tokio::runtime::blocking::pool::Inner::run
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/blocking/pool.rs:264:17
  29: tokio::runtime::blocking::pool::Spawner::spawn_thread::{{closure}}
             at /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.15.0/src/runtime/blocking/pool.rs:244:17
      std::sys_common::backtrace::__rust_begin_short_backtrace
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/sys_common/backtrace.rs:123:18
  30: std::thread::Builder::spawn_unchecked::{{closure}}::{{closure}}
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/thread/mod.rs:484:17
      <core::panic::unwind_safe::AssertUnwindSafe<F> as core::ops::function::FnOnce<()>>::call_once
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/panic/unwind_safe.rs:271:9
      std::panicking::try::do_call
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/panicking.rs:406:40
      std::panicking::try
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/panicking.rs:370:19
      std::panic::catch_unwind
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/panic.rs:133:14
      std::thread::Builder::spawn_unchecked::{{closure}}
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/thread/mod.rs:483:30
      core::ops::function::FnOnce::call_once{{vtable.shim}}
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/core/src/ops/function.rs:227:5
  31: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/alloc/src/boxed.rs:1694:9
      <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/alloc/src/boxed.rs:1694:9
      std::sys::unix::thread::Thread::new::thread_start
             at /rustc/02072b482a8b5357f7fb5e5637444ae30e423c40/library/std/src/sys/unix/thread.rs:106:17
  32: start_thread
  33: clone
```

**Comment by sunli829** [comment]
Thank! This backtrace is useful, and I already know roughly why.

**Comment by sunli829** [comment]
Fixed in `v1.2.42`

**Comment by Christoph-AK** [comment]
Thank you ❤️
